Source/Utils/WarpXUtil.cpp

```diff
@@ -349,6 +349,71 @@ void CheckGriddingForRZSpectral ()
+    }
+    // particle boundary may not be explicitly specified for some applications
+    bool particle_boundary_specified = false;
+    ParmParse pp("boundary");
+    pp.queryarr("field_lo", field_BC_lo, 0, AMREX_SPACEDIM);
+    pp.queryarr("field_hi", field_BC_hi, 0, AMREX_SPACEDIM);
+    if (pp.queryarr("particle_lo", particle_BC_lo, 0, AMREX_SPACEDIM))
+        particle_boundary_specified = true;
+    if (pp.queryarr("particle_hi", particle_BC_hi, 0, AMREX_SPACEDIM))
+        particle_boundary_specified = true;
+    AMREX_ALWAYS_ASSERT(field_BC_lo.size() == AMREX_SPACEDIM);
+    AMREX_ALWAYS_ASSERT(field_BC_hi.size() == AMREX_SPACEDIM);
+    AMREX_ALWAYS_ASSERT(particle_BC_lo.size() == AMREX_SPACEDIM);
+    AMREX_ALWAYS_ASSERT(particle_BC_hi.size() == AMREX_SPACEDIM);
+
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+        // Get field boundary type
+        WarpX::field_boundary_lo[idim] = GetBCTypeInteger(field_BC_lo[idim], true);
+        WarpX::field_boundary_hi[idim] = GetBCTypeInteger(field_BC_hi[idim], true);
+        // Get particle boundary type
+        WarpX::particle_boundary_lo[idim] = GetBCTypeInteger(particle_BC_lo[idim], false);
+        WarpX::particle_boundary_hi[idim] = GetBCTypeInteger(particle_BC_hi[idim], false);
+
+        if (WarpX::field_boundary_lo[idim] == FieldBoundaryType::Periodic ||
+            WarpX::field_boundary_hi[idim] == FieldBoundaryType::Periodic ||
+            WarpX::particle_boundary_lo[idim] == ParticleBoundaryType::Periodic ||
+            WarpX::particle_boundary_hi[idim] == ParticleBoundaryType::Periodic ) {
+            geom_periodicity[idim] = 1;
+            // to ensure both lo and hi are set to periodic consistently for both field and particles.
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+                (WarpX::field_boundary_lo[idim]  == FieldBoundaryType::Periodic) &&
+                (WarpX::field_boundary_hi[idim]  == FieldBoundaryType::Periodic),
+            "field boundary must be consistenly periodic in both lo and hi");
+            if (particle_boundary_specified) {
+                AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+                    (WarpX::particle_boundary_lo[idim] == ParticleBoundaryType::Periodic) &&
+                    (WarpX::particle_boundary_hi[idim] == ParticleBoundaryType::Periodic),
+               "field and particle boundary must be periodic in both lo and hi");
+            } else {
+                // set particle boundary to periodic
+                WarpX::particle_boundary_lo[idim] = ParticleBoundaryType::Periodic;
+                WarpX::particle_boundary_hi[idim] = ParticleBoundaryType::Periodic;
+                amrex::Warning("Particle boundary is set to periodic to be consistent with the fields.");
```

Comment:
I don't think that this warning is needed if this behavior is documented.

Reply:
Thank you @dpgrote . I added documentation and will remove this Warning.